File: fsharp/BronKerbosch/CliqueConsumer.fs
namespace BronKerbosch

open System.Diagnostics

type public CliqueConsumer =
    { MinSize: int
      Receiver: Clique -> Unit }

    member inline this.accept(clique: Clique) : Unit =
        Debug.Assert(clique.Length >= this.MinSize)
        this.Receiver clique

File: csharp/BronKerbosch/CliqueConsumer.cs
using System.Collections.Immutable;
using System.Diagnostics;

namespace BronKerbosch
{
    public sealed class CliqueConsumer<TAccumulator>(int minCliqueSize, TAccumulator Accu)
        where TAccumulator : ICliqueAccumulator<TAccumulator>
    {
        public bool IsAcceptedSize(int size) => size >= minCliqueSize;

        public void Accept(ImmutableArray<Vertex> clique)
        {
            Debug.Assert(IsAcceptedSize(clique.Length));
            Accu.Add(clique);
        }
    }
}
